ci/scripts/gen-flamegraph.sh

```diff
@@ -0,0 +1,264 @@
+  aws s3 rm s3://rw-ci-benchmark
+
+  echo ">>> Installing PromQL cli client"
+  # Download promql
+  wget https://github.com/nalbury/promql-cli/releases/download/v0.3.0/promql-v0.3.0-linux-amd64.tar.gz
+  tar -xvf promql-v0.3.0-linux-amd64.tar.gz
+  chmod +x ./promql
+  mv ./promql /usr/local/bin/promql
+  echo ">>> Run Sanity check that PromQL is installed"
+  promql --version
+
+  echo ">>> Installing Kafka"
+  wget https://downloads.apache.org/kafka/3.4.0/kafka_2.13-3.4.0.tgz
+  tar -zxvf kafka_2.13-3.4.0.tgz
+
+  echo ">>> Installing nexmark bench"
+  buildkite-agent artifact download nexmark-server /usr/local/bin
+  # Can't seem to do this in build phase.
+  chmod +x /usr/local/bin/nexmark-server
+
+  # TODO(kwannoel): eventually pin a newer version of nperf. This one works good enough for now.
+  echo ">>> Installing nperf"
+  wget https://github.com/koute/not-perf/releases/download/0.1.1/not-perf-x86_64-unknown-linux-gnu.tgz
+  tar -xvf not-perf-x86_64-unknown-linux-gnu.tgz
+}
+
+############## CONFIGURE
+
+configure_nexmark_bench() {
+mkdir -p nexmark-bench
+pushd nexmark-bench
+cat <<EOF > .env
+KAFKA_HOST="localhost:9092"
+BASE_TOPIC="nexmark"
+AUCTION_TOPIC="nexmark-auction"
+BID_TOPIC="nexmark-bid"
+PERSON_TOPIC="nexmark-person"
+NUM_PARTITIONS=8
+# NOTE: Due to https://github.com/risingwavelabs/risingwave/issues/6747, use `SEPARATE_TOPICS=false`
+SEPARATE_TOPICS=false
+RUST_LOG="nexmark_server=info"
+
+### DEFAULTS
+# host when running locally, use kafka1:19092 when running in docker
+# KAFKA_HOST="localhost:9092"
+# BASE_TOPIC="nexmark-events"
+# AUCTION_TOPIC="nexmark-auction"
+# BID_TOPIC="nexmark-bid"
+# PERSON_TOPIC="nexmark-person"
+# NUM_PARTITIONS=8
+# SEPARATE_TOPICS=true
+# RUST_LOG="nexmark_server=info"
+EOF
+popd
+}
+
+configure_rw() {
+# TODO(kwannoel): Workaround this error:
+# ```
+# Extracting dashboard artifacts to /risingwave/.risingwave/ui
+# fatal: detected dubious ownership in repository at '/risingwave'
+# ```
+git config --global --add safe.directory /risingwave
+pushd risingwave
+cat <<EOF > risedev-components.user.env
+RISEDEV_CONFIGURED=true
+
+ENABLE_PROMETHEUS_GRAFANA=true
+ENABLE_ETCD=true
+ENABLE_KAFKA=true
+ENABLE_COMPUTE_TRACING=true
+ENABLE_BUILD_RUST=true
+ENABLE_RELEASE_PROFILE=true
+ENABLE_ALL_IN_ONE=true
+EOF
+popd
+}
+
+configure_all() {
+  echo ">>> Configuring rw"
+  configure_rw
+  echo ">>> Configuring nexmark bench"
+  configure_nexmark_bench
+}
+
+############## Start benchmark environment
+
+# TODO(kwannoel): Currently we receive many errors for the dynamic libraries like so:
+# ```
+# major/minor/inode of /usr/lib/x86_64-linux-gnu/<some_library>.so.3" doesn't match the expected value: Some(Inode ...) != Some(Inode ...)
+# ```
+# This has minor effect on the flamegraph, so can ignore for now.
+# could it be related to profiling on Docker? Needs further investigation.
+start_nperf() {
+  ./nperf record -p `pidof compute-node` -o perf.data &
+}
+
+start_kafka() {
```

Comment:
Why don't we setup zk and kafka with risedev, but spawn them here separately?

Reply:
I will teardown the cluster for each new query I run, to ensure system is always in fresh state.

This also means `kafka` and `zookeeper` are torn down each time the cluster is restarted.

Then we will have to re-seed kafka events again.

Spawning them separately allows us to avoid this.

Reply:
Since the data in Kafka can be reentrant, I guess we can achieve this by creating a new command for RiseDev that only cleans the data of RisingWave nodes? 👀

Reply:
https://github.com/risingwavelabs/risingwave/issues/9437